File: AEDs2/Trabalho-Pratico-01/Palindromo/palindromo.c
#include <stdio.h>
#include <string.h>

/*
    Crie um método iterativo que recebe uma string como parâmetro e retorna true se essa é um palíndromo.


    Na saída padrão, para cada linha de entrada, escreva uma linha de saída com SIM/NAO indicando se a linha é um palíndromo.


    Destaca-se que uma linha de entrada pode ter caracteres não letras


    A entrada termina com a leitura de FIM
 */

int verifica(char x[]) {
    int tam = strlen(x);
    int vale = 1;

    for (int i = 0; i < tam; i++) {
        if (x[i] != x[tam - i - 1]) {
            vale = 0;
            break;
        }
    }

    return vale;
}

int main() {
    char x[500];  
    int res = 1;
    while (res == 1) {
        scanf("%[^\n]%*c", x);
    
        if (x[0] == 'F' && x[1] ==  'I' && x[2] == 'M')
        {
            res = 0;
        }
        else {
            int vale = verifica(x);
            if (vale == 1) {
                printf("SIM\n");
            } else {
                printf("NAO\n");
            }
        }
    }

    return 0;
}

File: AEDs2/Trabalho-Pratico-01/Palindromo/palindromo.java
import java.util.Scanner;

/*
    Crie um método iterativo que recebe uma string como parâmetro e retorna true se essa é um palíndromo.


    Na saída padrão, para cada linha de entrada, escreva uma linha de saída com SIM/NAO indicando se a linha é um palíndromo.


    Destaca-se que uma linha de entrada pode ter caracteres não letras


    A entrada termina com a leitura de FIM
 */

public class palindromo {

    public static boolean verifica(String x) {

        int tam = x.length();
        boolean vale = true;

        for (int i = 0; i < tam; i++) {
            if (x.charAt(i) != x.charAt(tam - i - 1)) {
                vale = false;
            }
        }

        if (vale == true) {
            return true;
        } else {
            return false;
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int res = 1;
        do {
            String x = sc.nextLine();
            boolean vale;

            if (x.compareTo("FIM") == 0) {
                res = 0;
            } else {
                vale = verifica(x);
                if (vale == true) {
                    System.out.println("SIM");
                } else {
                    System.out.println("NAO");
                }
            }
        } while (res == 1);
        sc.close();
    }
}